### PR TITLE
[PW-485] composer au jobs: extract and save commit SHA

### PR DESCRIPTION
**Pull request by alexdicianu**

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// I wanted to avoid a dependency loop by importing the jobs pkg (¯\_(ツ)_/¯).
+// Doesn't perform well for large slices, look at binary search or better algo for that.
+func InSlice[T comparable](needle T, haystack []T) bool {
+	for _, val := range haystack {
+		if val == needle {
+			return true
+		}
+	}
+	return false
+}
```
Comment (alexdicianu) [inline pkg/utils/utils.go:17]: Check out this bad boy 😄

Comment (alexdicianu) [inline pkg/utils/utils.go:17]: Check out this bad boy 😄

Comment (lloydjm77) [reply pkg/utils/utils.go:17]: I think this is our first use of generics.

**Review by threeiem — APPROVED** [review]
Nice! 👍

File: pkg/db/jobs.go [hunk 2]
```diff
@@ -168,3 +170,52 @@ func (q *internalQueryer) ListJobsByLabels(ctx context.Context, params map[strin
...
+			return nil // nothing to do.
+		}
+		data, err := json.Marshal(job.Spec.Labels)
+		if err != nil {
+			return fmt.Errorf("error marshaling labels: %v", err)
+		}
+		query := fmt.Sprintf(
+			`UPDATE job_manager.jobs SET spec['labels'] = '%s' WHERE job_id = $1;`,
+			string(data),
+		)
```
Comment (lloydjm77) [inline pkg/db/jobs.go:210]: You can do this directly in SQL like:

```
UPDATE job_manager.jobs SET spec['labels'] = spec['labels'] || '"dry_run.true"' WHERE job_id = '9b124030-fea0-45c9-89dd-3d8259d39789';
```

The `||` is a json array append operator in postgres.

Comment (lloydjm77) [reply pkg/db/jobs.go:210]: Just an FYI.

**Review by lloydjm77 — APPROVED** [review]

File: pkg/db/jobs.go (hunk 2 above)
Comment (lloydjm77) [reply pkg/db/jobs.go:210]: Yeah, there's all kinds of cool shit you can do with json in postgres.

**Pull request merged**